feat: add users page

Hiteazel opened the pull request:
Add users page with table and "add new user" button, with possibility of modifying/delete a user.

↗️ Created Files :
- UsersTable.tsx
- Users.tsx

➡️ Modified Files :
- types.ts
- routes.tsx
- Nav.tsx
- UserController.php
- RegisteredUserController.php

ADR1811 committed:
Update auth.php

ADR1811 committed:
Update auth.php

ADR1811 committed:
fix(register): Fillable user

ADR1811 committed:
fix(register): Fillable user

Hiteazel committed:
feat: add users page

Add users page with table and "add new user" button, with possibility of modifying/delete a user.

↗️ Created Files :
- UsersTable.tsx
- Users.tsx

→ Modified Files :
- types.ts
- routes.tsx
- Nav.tsx
- UserController.php
- RegisteredUserController.php

Hiteazel committed:
feat: add users page

Add users page with table and "add new user" button, with possibility of modifying/delete a user.

↗️ Created Files :
- UsersTable.tsx
- Users.tsx

→ Modified Files :
- types.ts
- routes.tsx
- Nav.tsx
- UserController.php
- RegisteredUserController.php

Hiteazel committed:
fix: remove onError

Hiteazel committed:
fix: remove onError

DestroyCom commented on sera-front/src/lib/types/types.ts:
```diff
@@ -17,3 +17,27 @@ export interface TicketsEntity {
   created_at: string;
   updated_at: string;
 }
+
+export interface Users {
+  pages?: (UsersEntity[] | null)[] | null;
+  pageParams?: null[] | null;
+}
+export interface UsersEntity {
+  id: number;
+  avatar_filename?: string;
+  avatar_url?: string;
+  firstname: string;
+  lastname: string;
+  email: string;
+  email_verified_at?: null;
+  role: string;
+  created_at?: string;
+  updated_at?: string;
+}
+
+export interface MyError {
+  message: string;
+  errors: {
+    [key: string]: string[];
+  };
+}
```
Ce fichier sera modifier en revoyant les types a l'avenir, mais cela passe dans cette PR

DestroyCom commented on sera-front/src/components/app/users/UsersTable.tsx:
```diff
@@ -0,0 +1,83 @@
+import { /* Check, */ /* Edit, */ Edit, Trash } from "lucide-react";
```
Enlever les imports commentés

DestroyCom commented on sera-back/app/Http/Controllers/Auth/RegisteredUserController.php:
```diff
@@ -46,7 +46,7 @@ public function store(Request $request): Response
         event(new Registered($user));
 
-        Auth::login($user);
+        // Auth::login($user);
```
@ADR1811 Normal que ce soit commenté ?

DestroyCom commented on sera-front/src/components/app/users/UsersTable.tsx:
```diff
@@ -0,0 +1,83 @@
+                  </p>
+                </TableCell>
+                <TableCell className="text-base text-black">
+                  <p className="w-12 truncate md:w-56">{user.email}</p>
```
Le champs se truncate ce serait bien de l'avoir en entier

DestroyCom replied:
![image](https://github.com/GoofyComponent/GoofySeRA/assets/37459465/76462f20-8354-423d-b2a6-ca9e6ddf0c1c)

DestroyCom commented on sera-front/src/pages/Users.tsx:
```diff
@@ -0,0 +1,626 @@
+  const [searchInput, setSearchInput] = useState("");
+  const navigate = useNavigate();
+  const { UserId } = useParams<{ UserId: string }>();
+  // const [previousMail, setPreviousMail] = useState("");
```
Commentaire a remove

DestroyCom replied:
Et a check si yen n'a pas d'autre

DestroyCom commented on sera-front/src/pages/Users.tsx:
```diff
@@ -0,0 +1,626 @@
+
+  useEffect(() => {
+    //DEBUG
+    console.log(setEditedUserData);
```
Remove ce console.log (et prob les autres)

DestroyCom commented on sera-front/src/pages/Users.tsx:
```diff
@@ -0,0 +1,626 @@
+              >
+                Search
+              </Button>
+              <Button
+                onClick={() => setUserDialogOpen(true)}
+                className="justify-end bg-sera-jet text-sera-periwinkle hover:bg-sera-jet/50 hover:text-sera-periwinkle/50"
+              >
+                Add User
+              </Button>
```
Plus rapprocher le bouton du coté droit

DestroyCom replied:
Intervertir les champs de filtre avec le bouton add, en gros :
Mtn:
filtre - adduser
voulu :
adduser - filtre

DestroyCom commented on sera-front/src/pages/Users.tsx:
```diff
@@ -0,0 +1,626 @@
+            open={userDialogOpen}
+          >
+            <div className="mx-6 flex justify-between text-4xl font-semibold text-sera-jet">
+              <Input
+                className="mr-2 w-[360px]"
+                type="text"
+                placeholder="Search (Firstname, Lastname, Email)"
+                value={searchInput}
+                onChange={(e) => setSearchInput(e.target.value)}
+              />
```
le champs fonctionne très bien mais il faut trouver une solution pour pouvoir remettre tout les utilisateurs apres avoir chercher via texte

DestroyCom commented on sera-front/src/pages/Users.tsx:
```diff
@@ -0,0 +1,626 @@
+            <div>
+              <div className="flex flex-col">
+                <Label htmlFor="firstname">Firstname</Label>
+                <Input
+                  type="text"
+                  id="firstname"
+                  value={UserData.firstname}
+                  placeholder={editedUserData?.firstname}
+                  className="col-span-3"
+                  onChange={(e) =>
+                    setUserData({
+                      ...UserData,
+                      firstname: e.target.value,
+                    })
+                  }
+                />
```
Ajouter des espacement Y entre le input et le label pr que ce soit tjr lisible avec le outline

DestroyCom reviewed (changes requested):
C'est globalement bon & le code est valide/compréhensible

Mais ya deux trois trucs a change 👍🏻

Hiteazel committed:
feat: Ease of use User Search

⭐Search bar in Users : When it's empty, shows all users without pressing search button

🛠️Delete unused impor, console.log and comment
🛠️Table column size fix
🛠️Swap place of the buttons "Add User" and "Search"
🛠️Adding spaces in forms for more visibility

Hiteazel committed:
feat: Ease of use User Search

⭐Search bar in Users : When it's empty, shows all users without pressing search button

🛠️Delete unused impor, console.log and comment
🛠️Table column size fix
🛠️Swap place of the buttons "Add User" and "Search"
🛠️Adding spaces in forms for more visibility

Hiteazel committed:
fix: Remove line auto log in when registered

🛠Remove line auto log in when registered (validated by @ADR1811 )

Hiteazel committed:
fix: Remove line auto log in when registered

🛠Remove line auto log in when registered (validated by @ADR1811 )

Hiteazel committed:
fix: Change title div

🛠️Fix button "Search" to right
🛠️Swap SearchInput and FilterRole
🛠️Move "Add User" button to left

Hiteazel committed:
Merge branch 'SERA-30' of https://github.com/GoofyComponent/GoofySeRA into SERA-30

Hiteazel committed:
feat: Display roles table

🛠️Display formatted roles in the table instead of value getted
